driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -190,13 +192,14 @@ public void getAsync(final OperationContext operationContext, final SingleResult
         }
         connectionPoolListener.connectionCheckOutStarted(new ConnectionCheckOutStartedEvent(serverId));
         Timeout timeout = Timeout.startNow(settings.getMaxWaitTime(NANOSECONDS));
-        SingleResultCallback<InternalConnection> eventSendingCallback = (result, failure) -> {
+        SingleResultCallback<PooledConnection> eventSendingCallback = (connection, failure) -> {
             SingleResultCallback<InternalConnection> errHandlingCallback = errorHandlingCallback(callback, LOGGER);
             if (failure == null) {
-                connectionPoolListener.connectionCheckedOut(new ConnectionCheckedOutEvent(getId(result)));
-                errHandlingCallback.onResult(result, null);
+                connection.checkedOutForOperation(operationContext);
+                connectionPoolListener.connectionCheckedOut(new ConnectionCheckedOutEvent(getId(connection)));
```

Comment:
Done.  I missed these because a) I was careless, and b) we're suppressing deprecation warnings because of all the deprecated event types that we still use here.